qiskit/quantum_info/operators/symplectic/base_pauli.py

```diff
@@ -0,0 +1,639 @@
+"""
+# pylint: disable=invalid-name, abstract-method
+
+import copy
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.barrier import Barrier
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+
+class BasePauli(BaseOperator):
+    r"""Symplectic representation of a list of N-qubit Paulis.
+
+    Base class for Pauli and PauliList.
+    """
+
+    def __init__(self, z, x, phase):
+        """Initialize the BasePauli.
+
+        This is an array of M N-qubit Paulis defined as
+        P = (-i)^phase Z^z X^x.
+
+        Args:
+            z (array): input z matrix.
+            x (array): input x matrix.
+            phase (array): input phase vector.
+        """
+        self._z = z
+        self._x = x
+        self._phase = phase
+        self._num_paulis, num_qubits = self._z.shape
+        super().__init__(num_qubits=num_qubits)
+
+    def copy(self):
+        """Make a deep copy of current operator."""
+        # Deepcopy has terrible performance on objects with Numpy arrays
+        # attributes so we make a shallow copy and then manually copy the
+        # Numpy arrays to efficiently mimic a deepcopy
+        ret = copy.copy(self)
+        ret._z = self._z.copy()
+        ret._x = self._x.copy()
+        ret._phase = self._phase.copy()
+        return ret
+
+    # ---------------------------------------------------------------------
+    # BaseOperator methods
+    # ---------------------------------------------------------------------
+
+    def tensor(self, other):
+        """Return the tensor product Pauli self ⊗ other.
+
+        Args:
+            other (BasePauli): another Pauli.
+
+        Returns:
+            BasePauli: the tensor product Pauli.
+        """
+        z = np.hstack([self._stack(other._z, self._num_paulis), self._z])
+        x = np.hstack([self._stack(other._x, self._num_paulis), self._x])
+        phase = np.mod(self._phase + other._phase, 4)
+        return BasePauli(z, x, phase)
+
+    def expand(self, other):
+        """Return the tensor product Pauli other ⊗ self.
+
+        Args:
+            other (BasePauli): another Pauli.
+
+        Returns:
+            BasePauli: the tensor product Pauli.
+        """
+        z = np.hstack([self._z, self._stack(other._z, self._num_paulis)])
+        x = np.hstack([self._x, self._stack(other._x, self._num_paulis)])
+        phase = np.mod(self._phase + other._phase, 4)
+        return BasePauli(z, x, phase)
+
+    # pylint: disable=arguments-differ
+    def compose(self, other, qargs=None, front=True, inplace=False):
+        """Return the composition of Paulis self∘other.
+
+        Note that this discards phases.
+
+        Args:
+            other (BasePauli): another Pauli.
+            qargs (None or list): qubits to apply dot product on (default: None).
+            front (bool): If True use `dot` composition method (default: False).
```

Comment:
This lines states that `default: False`, but the default is set `True` in Line 93. Which is correct?

Reply:
Good catch, it should be False